Unity3D/Assets/RosSharp/Editor/UrdfImporter/UrdfRobotExtensions.cs

```diff
@@ -59,5 +78,93 @@ public static void setKinematic(GameObject robot, bool isKinematic)
             foreach (Rigidbody rigidbody in rigidbodies)
                 rigidbody.isKinematic = isKinematic;
         }
+
+        public static bool ImportUrdfToUnity(Robot robot)
+        {
+            // Create URDF folder
+            if (!AssetDatabase.IsValidFolder("Assets/Urdf"))
+            {
+                AssetDatabase.CreateFolder("Assets", "Urdf");
+            }
+
+            // Create folder for URDF
+            if (!AssetDatabase.IsValidFolder("Assets/Urdf/" + robot.name))
+            {
+                AssetDatabase.CreateFolder("Assets/Urdf", robot.name);
+            }
+            // TODO: if folder exists, check if robot should be updated.
+
+            // Create folder for meshes
+            if (!AssetDatabase.IsValidFolder("Assets/Urdf/" + robot.name + "/meshes"))
+            {
+                AssetDatabase.CreateFolder("Assets/Urdf/" + robot.name, "meshes");
+            }
+
+            // Copy URDF into folder
+            string unity_urdf_file_path = Path.Combine(Application.dataPath + "/Urdf/" + robot.name, robot.name + ".urdf");
+            unity_urdf_file_path = unity_urdf_file_path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+
+            // Find the root of the description package
```

Comment:
In the following lines (107-120) you are searching for a package.xml.
This should must be put in a separate method.
Why dont you use [Directory.GetFile](https://msdn.microsoft.com/de-de/library/wz42302f(v=vs.110).aspx) for it?

Reply:
I did it this way because the user only selects the URDF and I don't know how many directories to go back up to look for the package.xml, with Directory.GetFiles you'd have to ask the user to also point to the package root right? Is there a better way to do this without asking the user for too much information?